src/components/CastDetails/CastDetails.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import '../CastDetails/CastDetails.scss'
+
+interface CastMemberDetails {
+  id: number;
+  title: string;
+  character: string;
+  poster_path: string;
+  release_date: string;
+  media_type: 'movie' | 'tv';
+}
+
+const IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w500"; // Base URL for images
+
+const CastDetails = () => {
+  const { castId } = useParams<{ castId: string }>();
+  const [movies, setMovies] = useState<CastMemberDetails[]>([]);
+  const [tvShows, setTvShows] = useState<CastMemberDetails[]>([]);
+
+  useEffect(() => {
+    const fetchCastDetails = async () => {
+      try {
+        const apiKey = "f1a02268af3a2e076dc84ca1a6aaaefe";
+        const response = await axios.get(
+          `https://api.themoviedb.org/3/person/${castId}/combined_credits?api_key=${apiKey}&language=en-US`
+        );
+        console.log("Cast details:", response.data);
+
+        const castData: CastMemberDetails[] = response.data.cast;
+
+        const moviesData = castData.filter((item: CastMemberDetails) => item.media_type === 'movie');
+        const tvShowsData = castData.filter((item: CastMemberDetails) => item.media_type === 'tv');
+
+        setMovies(moviesData);
+        setTvShows(tvShowsData);
+      } catch (error) {
+        console.error("Error fetching cast details:", error);
+      }
+    };
+
+    fetchCastDetails();
+  }, [castId]);
+
+  return (
+    <div>
+      <div>
+        <h2>Movies</h2>
+        <div className="moviesList">
+          {movies.length ? (
+            movies.map((movie) => (
+              <div key={movie.id}>
```

Comment:
please extract this div into a separate component (MovieCard) and use this component for movies and tvShows